File: ios/MenuMaker/Data/Models/FavoriteModels.swift
import Foundation

// MARK: - Favorite Models

struct Favorite: Codable, Identifiable {
    let id: String
    let userId: String
    let businessId: String
    let business: Business?
    let createdAt: String

    var formattedDate: String {
        guard let date = ISO8601DateFormatter().date(from: createdAt) else {
            return createdAt
        }

        let formatter = DateFormatter()
        formatter.dateStyle = .medium
        return formatter.string(from: date)
    }
}

struct FavoriteResponse: Decodable {
    let success: Bool
    let data: FavoriteData
}

struct FavoriteData: Decodable {
    let favorite: Favorite
}

struct FavoriteListResponse: Decodable {
    let success: Bool
    let data: FavoriteListData
}

struct FavoriteListData: Decodable {
    let favorites: [Favorite]
}

struct AddFavoriteRequest: Encodable {
    let businessId: String
}

struct RemoveFavoriteRequest: Encodable {
    let businessId: String
}

File: android/app/src/main/kotlin/com/menumaker/data/remote/models/FavoriteModels.kt
package com.menumaker.data.remote.models

import com.google.gson.annotations.SerializedName

/**
 * Data Transfer Object for Favorite
 * Represents a user's favorited business/seller
 */
data class FavoriteDto(
    @SerializedName("id") val id: String,
    @SerializedName("user_id") val userId: String,
    @SerializedName("business_id") val businessId: String,
    @SerializedName("business") val business: BusinessDto?,
    @SerializedName("created_at") val createdAt: String
)

/**
 * API Response for a single favorite
 */
data class FavoriteResponse(
    @SerializedName("success") val success: Boolean,
    @SerializedName("data") val data: FavoriteData
)

data class FavoriteData(
    @SerializedName("favorite") val favorite: FavoriteDto
)

/**
 * API Response for a list of favorites
 */
data class FavoriteListResponse(
    @SerializedName("success") val success: Boolean,
    @SerializedName("data") val data: FavoriteListData
)

data class FavoriteListData(
    @SerializedName("favorites") val favorites: List<FavoriteDto>
)

/**
 * Request to add a business to favorites
 */
data class AddFavoriteRequest(
    @SerializedName("business_id") val businessId: String
)

/**
 * Request to remove a business from favorites
 */
data class RemoveFavoriteRequest(
    @SerializedName("business_id") val businessId: String
)
